### ISSUE TITLE
Get along better with other agents

### ISSUE THREAD

**Issue by gihad** [glowroot/glowroot#321]
When adding glowroot to an application that is already instrumented with the Newrelic agent, the application will crash.

```
scoreboard_1  | 2018-02-08 21:43:40.133 INFO  org.glowroot - Glowroot version: 0.10.2, built 2018-02-05 04:02:53 +0000
scoreboard_1  | 2018-02-08 21:43:40.143 INFO  org.glowroot - Java version: 1.8.0_121
scoreboard_1  | 2018-02-08 21:43:48.302 INFO  org.glowroot - plugins loaded: Cassandra Plugin, Elasticsearch Plugin, Executor Plugin, Grails Plugin, Hibernate Plugin, HTTP Client Plugin, Java HTTP Server Plugin, JAX-RS Plugin, Jdbc Plugin, JMS Plugin, JSF Plugin, JSP Plugin, Logger Plugin, Play Plugin, Netty Plugin, Quartz Plugin, Redis Plugin, Servlet Plugin, Spring Plugin, Struts Plugin
scoreboard_1  | 2018-02-08 21:43:50.322 INFO  org.glowroot - creating glowroot schema ...
scoreboard_1  | 2018-02-08 21:43:54.968 INFO  org.glowroot - glowroot schema created
scoreboard_1  | 2018-02-08 21:43:57.583 INFO  org.glowroot - UI listening on 0.0.0.0:4000
scoreboard_1  | Feb 8, 2018 21:43:59 +0000 [1 1] com.newrelic INFO: New Relic Agent: Loading configuration file "./newrelic.yml"
scoreboard_1  | Feb 8, 2018 21:44:02 +0000 [1 1] com.newrelic INFO: New Relic Agent: Writing to log file: /logs/newrelic_agent.log
scoreboard_1  | Unable to start New Relic Agent. Please remove -javaagent from your startup arguments and contact New Relic support.
scoreboard_1  | java.lang.UnsupportedOperationException: class redefinition failed: attempted to change superclass or interfaces
scoreboard_1  | 	at sun.instrument.InstrumentationImpl.$$YJP$$retransformClasses0(Native Method)
scoreboard_1  | 	at sun.instrument.InstrumentationImpl.retransformClasses0(InstrumentationImpl.java)
scoreboard_1  | 	at sun.instrument.InstrumentationImpl.retransformClasses(InstrumentationImpl.java:144)
scoreboard_1  | 	at com.newrelic.agent.util.InstrumentationWrapper.retransformClasses(InstrumentationWrapper.java:49)
scoreboard_1  | 	at com.newrelic.agent.instrumentation.ClassTransformerServiceImpl.retransformMatchingClassesImmediately(ClassTransformerServiceImpl.java:229)
scoreboard_1  | 	at com.newrelic.agent.instrumentation.weaver.ClassWeaverService$RetransformRunnable.run(ClassWeaverService.java:429)
scoreboard_1  | 	at com.newrelic.agent.instrumentation.context.InstrumentationContextManager.create(InstrumentationContextManager.java:281)
scoreboard_1  | 	at com.newrelic.agent.instrumentation.ClassTransformerServiceImpl.startClassTransformer(ClassTransformerServiceImpl.java:156)
scoreboard_1  | 	at com.newrelic.agent.instrumentation.ClassTransformerServiceImpl.doStart(ClassTransformerServiceImpl.java:114)
scoreboard_1  | 	at com.newrelic.agent.service.AbstractService.start(AbstractService.java:56)
scoreboard_1  | 	at com.newrelic.agent.service.ServiceManagerImpl.doStart(ServiceManagerImpl.java:146)
scoreboard_1  | 	at com.newrelic.agent.service.AbstractService.start(AbstractService.java:56)
scoreboard_1  | 	at com.newrelic.agent.Agent.premain(Agent.java:286)
scoreboard_1  | 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
scoreboard_1  | 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
scoreboard_1  | 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
scoreboard_1  | 	at java.lang.reflect.Method.invoke(Method.java:498)
scoreboard_1  | 	at com.newrelic.bootstrap.BootstrapAgent.startAgent(BootstrapAgent.java:130)
scoreboard_1  | 	at com.newrelic.bootstrap.BootstrapAgent.premain(BootstrapAgent.java:79)
scoreboard_1  | 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
scoreboard_1  | 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
scoreboard_1  | 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
scoreboard_1  | 	at java.lang.reflect.Method.invoke(Method.java:498)
scoreboard_1  | 	at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:386)
scoreboard_1  | 	at sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:401)
leaderboard_scoreboard_1 exited with code 1
```

**Comment by trask** [comment]
Hi @gihad, thanks for reporting this! I think it should be fixed now in the latest snapshot release [glowroot-agent-0.10.4-SNAPSHOT-dist.zip](https://oss.sonatype.org/service/local/artifact/maven/redirect?r=snapshots&g=org.glowroot&a=glowroot-agent&c=dist&e=zip&v=0.10.4-SNAPSHOT).

**Comment by trask** [comment]
I posted too soon, the building was still running, but it is available now [glowroot-agent-0.10.4-SNAPSHOT-dist.zip](https://oss.sonatype.org/service/local/artifact/maven/redirect?r=snapshots&g=org.glowroot&a=glowroot-agent&c=dist&e=zip&v=0.10.4-SNAPSHOT).
